src/main/java/com/metamx/metrics/JvmMonitor.java

```diff
@@ -33,31 +34,65 @@
 
 public class JvmMonitor extends AbstractMonitor
 {
-  /**
+  /*
    * The following GC-related code is partially based on
    * https://github.com/aragozin/jvm-tools/blob/e0e37692648951440aa1a4ea5046261cb360df70/
    * sjk-core/src/main/java/org/gridkit/jvmtool/PerfCounterGcCpuUsageMonitor.java
    */
 
   private enum GcGeneration
   {
-    YOUNG(GarbageCollectors.youngName(), 0), OLD(GarbageCollectors.oldName(), 1);
+    YOUNG(0) {
+      @Override
+      String readableGcName(String name)
+      {
+        if (name.contains("Copy")) {
```

Comment:
Sorry, didn't get your point. Ok, I understand that some of them can contain common part in the name, but they will still have some differences, as far as I understand. For example here is the list of GC names for my local machine with different GC settings:
```
-XX:+UseSerialGC
    sun.gc.collector.0.name (PerfStringCounter): Copy
    sun.gc.collector.1.name (PerfStringCounter): MSC

-XX:+UseParallelGC
    sun.gc.collector.0.name (PerfStringCounter): PSScavenge
    sun.gc.collector.1.name (PerfStringCounter): PSParallelCompact

-XX:+UseConcMarkSweepGC
    sun.gc.collector.0.name (PerfStringCounter): PCopy
    sun.gc.collector.1.name (PerfStringCounter): CMS

-XX:+UseG1GC
    sun.gc.collector.0.name (PerfStringCounter): G1 incremental collections
    sun.gc.collector.1.name (PerfStringCounter): G1 stop-the-world full collections
```

You can see here `PCopy` and `Copy`. It's different GC, but current implementation will use the same name `copy` for both of them. Also for Concurrent Mark and Sweep `sun.gc.collector.1.name` contains just CMS, but current code expects `ConcurrentMarkSweep` as part of the name. So, my suggestion is pass `sun.gc.collector.0.name` and `sun.gc.collector.1.name` counter's values as is in `gcName` dimension.